subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrading_version_8]]
+= Upgrading your build from Gradle 8.x to the latest
+
+This chapter provides the information you need to migrate your Gradle 8.x builds to the latest Gradle release. For migrating from Gradle 4.x, 5.x, 6.x, or 7.x, see the <<upgrading_version_7.adoc#upgrading_version_7, older migration guide>> first.
+
+We recommend the following steps for all users:
+
+. Try running `gradle help --scan` and view the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan.
++
+image::deprecations.png[Deprecations View of a Gradle Build Scan]
++
+This is so that you can see any deprecation warnings that apply to your build.
++
+Alternatively, you could run `gradle help --warning-mode=all` to see the deprecations in the console, though it may not report as much detailed information.
+. Update your plugins.
++
+Some plugins will break with this new version of Gradle, for example because they use internal APIs that have been removed or changed. The previous step will help you identify potential problems by issuing deprecation warnings when a plugin does try to use a deprecated part of the API.
++
+. Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
+. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.
+
+[[changes_9.0]]
+== Upgrading from 8.x and earlier
+
+=== Warnings that are now errors
+
+[[deprecated_configurations_should_not_be_used]]
+==== Usage of deprecated configurations is now an error
+
+Usage of the following `@link:{javadocPath}/javadoc/org/gradle/api/internal/artifacts/configurations/ConfigurationRoles.htm[ConfigurationRoles]` are now errors:
```

Comment:
I just wanted to add something so that the links in the deprecation messages would go somewhere, and didn't put too much effort into these yet.  I'll come back and revise them after we finalize these change.  I thought the new upgrading guide page wouldn't be visible unless it is linked?

Reply:
No idea.  I was more curious whether this was something we're doing now as a practice or if this was a one-off.  If the plan is to circle back on this, I don't think there's anything that needs to be resolved now.

Reply:
Do we plan to update this now that we're closer to merging?